core/src/main/scala/org/apache/spark/scheduler/ActiveJob.scala

```diff
@@ -60,4 +60,10 @@ private[spark] class ActiveJob(
   val finished = Array.fill[Boolean](numPartitions)(false)
 
   var numFinished = 0
+
+  // Mark all the partitions of the stage to be not finished.
+  def clearResult(): Unit = {
```

Comment:
It is not clear to me that "clearResult" accurately describes what it does. `markAllPartitionsAsUnfinished`? Better names are welcome!